### PR TITLE
Update CodonTable.py and Restriction_Dictionary.py for new release.

### LINKED ISSUE #2080: Codon table is out of date
[`gc.prt`](ftp://ftp.ncbi.nih.gov/entrez/misc/data/gc.prt), from which the `Data/CodonTable` file is generated, now includes codon tables up to 32 (the [site](https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi#SG33) goes up to 33, though).

Can we get it updated?

### PR DESCRIPTION
**Pull request by MarkusPiotrowski**
This pull request addresses issue #2080. It updates `CodonTable.py` to the recent codon table version 4.4 from [ftp://ftp.ncbi.nih.gov/entrez/misc/data/gc.prt](url).

Also, `Restriction_Dictionary` was updated to Rebase version 905 (May 2019)

This PR will close #2080.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Tests/test_CodonTable.py [hunk 1]
```diff
@@ -159,7 +159,7 @@ def test_stop_codons(self):
 
             # R = A or G, so URR = UAA or UGA / TRA = TAA or TGA = stop codons
             if "UAA" in amb_rna.stop_codons and "UGA" in amb_rna.stop_codons \
-               and id != 28:
+               and id not in(28, 32):
```
Comment (peterjc) [inline Tests/test_CodonTable.py]: needs a space there, flake8 will likely spot this too.

File: Tests/test_CodonTable.py [hunk 2]
```diff
@@ -667,6 +668,29 @@ def test_table31(self):
...
         self.assertEqual(nuc_table.forward_table["UGA"], "W")
 
+    def test_table32(self):
+        """Check table 32: Balanophoraceae Plastid.
+
+        NOTE: This code has one unambiguous stop codon! TAA codes for either
+        stop or W, dependent on the context.
+        Another note: I guess this is wrong. There is a recent publication
+        that shows that TAA is not a stop codon anymore but only coding for
+        W. So table 32 may change in the near future.git
```
Comment (peterjc) [inline Tests/test_CodonTable.py]: Is the "git" here from typing in the wrong window, or a tongue in cheek comment?

Comment (MarkusPiotrowski) [reply Tests/test_CodonTable.py]: :-(, yes, together with a trailing whitespace....

Comment (MarkusPiotrowski) [reply Tests/test_CodonTable.py]: I introduced this while removing a whitespace issue, so the new whitespace warning during the commit passed my attention....

File: Tests/test_CodonTable.py (hunk 1 above)
Comment (MarkusPiotrowski) [reply Tests/test_CodonTable.py]: Unfortunately, flake8 didn't flag this.

Comment (peterjc) [reply Tests/test_CodonTable.py]: You're right. I wonder if it assumes from the context that ``in(28, 32)`` is calling a function? Logged as https://github.com/PyCQA/pycodestyle/issues/843#issuecomment-496212488 but see also https://github.com/PyCQA/pycodestyle/issues/631

**Comment by peterjc** [comment]
Looks good to merge (AppVeyor still running)

File: Tests/test_CodonTable.py (hunk 1 above)
Comment (peterjc) [reply Tests/test_CodonTable.py]: Found a better matched issue, https://github.com/PyCQA/pycodestyle/issues/646

**Pull request merged**